### ISSUE TITLE
Search results in SearchView sample does not respond when clicking.

### ISSUE THREAD

**Issue by m-pact** [rokudev/SceneGraphDeveloperExtensions#16]
Nothing happens when clicking on the search results in SearchView from the sample.  Shouldn't it take you to a node?

**Comment by RokuChris** [comment]
This is intentional. The sample prints some output to the console when the user selects an item. This happens in the `OnSearchItemSelected()` observer function. In a real world channel, a developer could replace that with whatever logic they needed for their particular use case.

**Comment by m-pact** [comment]
In SGDEX SearchView 'OnSearchItemSelected()' is not an option on the CHSearch.brs page.  I assume it is in the actually SGDEX extention.  I'm not sure we should change that.

The searchView works great all the way up to the when you click on the results.  There results are there.  But doesn't take you anywhere.

it would be nice to have a full working example to build from.

Thanks, M-pact

**Comment by RokuChris** [comment]
`CHSearch.brs` is a ContentHandler. ContentHandlers provide the data to populate a view. They don't react to user actions.

To handle a user's selection in SearchView, you need to observe the _rowItemSelected_ field on the view. In the SearchView sample, this is done in `MainScene.brs`. 

```
    searchView = CreateObject("roSGNode", "SearchView")
    searchView.hintText = "Enter search term"
    ' query field will be changed each time user has typed something
    searchView.ObserveFieldScoped("query", "OnSearchQuery")
    searchView.ObserveFieldScoped("rowItemSelected", "OnSearchItemSelected")

```

**Comment by m-pact** [comment]
Oh, ok.  I think I got it now.

So I need to create a new sub for "OnSearchItemSelected"

Then emulate the gridview if I want the results to behave like a gridview.

Something like below?

sub OnSearchItemSelected(event as Object)
    grid = event.GetRoSGNode()
    selectedIndex = event.Getdata()
    rowContent = grid.content.GetChild(selectedIndex[0])
    detailsView = ShowDetailsView(rowContent, selectedIndex[1])
    detailsView.ObserveField("wasClosed", "OnDetailsWasClosed")
end sub

**Comment by m-pact** [comment]
Yep, that was it.  Got it figured.  Thanks for the response RokuChris.  This really isn't an issue.  It was a learning issue on my side.  Thanks, M-pact
